pwnlib/gdb.py

```diff
@@ -458,6 +522,13 @@ def debug(args, gdbscript=None, exe=None, ssh=None, env=None, sysroot=None, api=
 
         >>> io.interactive() # doctest: +SKIP
         >>> io.close()
+        
+        Start a new process with modified argv[0]
+        >>> io = gdb.debug("/bin/sh", 'continue', argv=[b'\xde\xad\xbe\xef'])
```

Comment:
This usage is weird, since `gdb.debug` doesn't have an `argv` parameter but only an `args` one and this appears to rely on `process` having an `argv` parameter? I'd expect this to look like `gdb.debug([b'\xde\xad\xbe\xef'], 'continue', exe="/bin/sh")`. Same for the other added examples.

I'm confused on how this works with `gdbserver = runner(args, env=env, aslr=1, **kwargs)` when `runner` is `tubes.process.process`. The first argument is argv and the `kwargs` contain the explicit `argv` argument again, no? Which one is used?

Reply:
Fixed in commit `gdb.py: Add test case for LD_Preload`

Reply:
No, there are still calls of `gdb.debug` with an `argv` argument. There is no `argv`, only `args` in `gdb.debug`.